gun.js

```diff
@@ -130,20 +130,21 @@
 	})(USE, './onto');
 
 	;USE(function(module){
-		USE('./shim');
-		module.exports = function(v){ // Valid values are a subset of JSON: null, binary, number (!Infinity), text, or a soul relation. Arrays need special algorithms to handle concurrency, so they are not supported directly. Use an extension that supports them if needed but research their problems first.
-			if(v === undefined){ return false }
-			if(v === null){ return true } // "deletes", nulling out keys.
-			if(v === Infinity){ return false } // we want this to be, but JSON does not support it, sad face.
-			if(v !== v){ return false } // can you guess what this checks for? ;)
-			if('string' == typeof v // text!
-			|| 'boolean' == typeof v
-			|| 'number' == typeof v){
-				return true; // simple values are valid.
-			}
-			if(v && ('string' == typeof (v['#']||0)) && Object.empty(v, ['#'])){ return v['#'] } // is link
-			return false; // If not, everything else remaining is an invalid data type. Custom extensions can be built on top of these primitives to support other types.
-		}
+		// Valid values are a subset of JSON: null, binary, number (!Infinity), text,
+		// or a soul relation. Arrays need special algorithms to handle concurrency,
+		// so they are not supported directly. Use an extension that supports them if
+		// needed but research their problems first.
+		module.exports = (v) =>
+			// erases
+			v === undefined ||
+			// "deletes", nulling out keys.
+			v === null ||
+			"string" === typeof v ||
+			"boolean" === typeof v ||
+			// we want +/- Infinity to be, but JSON does not support it, sad face.
+			// can you guess what v === v checks for? ;)
+			("number" === typeof v && v != Infinity && v != -Infinity && v === v) ||
+			(v && "string" == typeof v["#"] && Object.keys(v).length === 1 && v["#"]);
```

Comment:
some inline comments via git here would be helpful to know/understand what is changing.

Reply:
Here we add `undefined` as an option and ban `-Infinity`.
The logic is as simple as:
- if it is `undefined` or `null` or `string` or `boolean` or `number` (which is not `+/- Infinity`, nor `NaN`), return `true`
- if it is an object of format `{ '#' }`, it's a soul, and we return it's `#` value
- otherwise, we return false